apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -66,25 +70,30 @@ class SOMATileDBContext(ContextBase):
 
     def __init__(
         self,
-        tiledb_config: Optional[Dict[str, Union[str, float]]] = None,
-        timestamp: Optional[OpenTimestamp] = None,
-        threadpool: Optional[ThreadPoolExecutor] = None,
+        tiledb_config: Dict[str, Union[str, float]] | None = None,
```

Comment:
the `Union[foo,bar]` and the `foo | bar` syntax are equivalent.  Recommend picking one, e.g., either:

> Dict[str, str|float] | None

or

> Union[Dict[str, Union[str, float]], None]

The first one is the more modern syntax and (I think) is more readable.